src/idl_gen_swift.cpp

```diff
@@ -590,9 +591,10 @@ class SwiftGenerator : public BaseGenerator {
     code_.SetValue("VALUETYPE", type);
     code_.SetValue("OFFSET", name);
     code_.SetValue("CONSTANT", field.value.constant);
-    std::string nullable = field.optional ? "nil" : "{{CONSTANT}}";
-    std::string optional = field.optional ? "?" : "";
-    auto const_string = "return o == 0 ? " + nullable + " : ";
+    bool opt_scalar = field.optional && IsScalar(field.value.type.base_type);
```

Comment:
Something about optional-vectors broke... so I changed the condition to optional scalars. @mustiikhalil may understand better

Reply:
@CasperN from what I can see now both, the vector count and vector element are considered optional. Is that a change that we want to have, since `optional_vector: [int32] = null;` doesn't work? other than that, the changes seem to be fine.
affected part:
```swift
/// `return o == 0 ? nil` was being generated instead of `return o == 0 ? 0`, thus the errors.
/// however I still don't understand why inventory would be optional in `field.optional`
public var inventoryCount: Int32 { let o = _accessor.offset(VTOFFSET.inventory.v); return o == 0 ? 0 : _accessor.vector(count: o) }
public func inventory(at index: Int32) -> UInt8 { let o = _accessor.offset(VTOFFSET.inventory.v); return o == 0 ? 0 : _accessor.directRead(of: UInt8.self, offset: _accessor.vector(at: o) + index * 1) }
    
```

Reply:
`optional_vector: [int32] = null` doesn't work but I thought vectors were currently optional already. At least, I think they are in C++ and Rust. Part of the idea in #6053 was to give schema-writers a way of ignoring the difference between a non-present vector and a present zero-length vector.

Reply:
Well, I believe that's just for rust and cpp. But I'm not sure, since java can return null without explicitly stating that it's optional

Reply:
@aardappel overall, I think we're good on this PR, but this does introduce some complexity for #6053 since the semantics of optional vs required for vectors is already inconsistent between languages.

Reply:
all non-scalar types default to optional/null in all languages so far, yes.

Reply:
Unless I'm misinterpreting, it seems Swift's vectors default to empty, not null.